src/lib/gssapi/krb5/acquire_cred.c

```diff
@@ -522,7 +525,11 @@ get_name_from_client_keytab(krb5_context context, krb5_gss_cred_id_rec *cred)
     krb5_principal princ;
 
     assert(cred->name == NULL);
-    code = k5_kt_get_principal(context, cred->client_keytab, &princ);
+    if (cred->client_keytab) {
```

Comment:
I would just add "if (cred->client_keytab == NULL) return KRB5_KT_NOTFOUND;" here and not change the existing code.